papermill/tests/test_gcs.py

```diff
@@ -42,7 +47,20 @@ def test_gcs_write(mocker):
     gcs_handler = GCSHandler()
     client = gcs_handler._get_client()
     gcs_handler.write('new value', 'gs://bucket/test.ipynb')
+    # Check that client is only generated once
+    assert client is gcs_handler._get_client()
+
 
+def test_gcs_write_retry(mocker):
+    mocker.patch('papermill.iorw.GCSFileSystem', MockGCSFileSystem)
+    counter = 0
+    try:
+        gcs_handler = GCSHandler()
+        client = gcs_handler._get_client()
+        gcs_handler.write('', 'gs://bucket/test.ipynb')
+    except RetryableError:
```

Comment:
I don't think this actually tests the retry code path. RetryableError isn't being caught by the handler and automatically rerun because it's not a RateLimitException.

I think there should be one test which tests that retries can be exhausted by returning RateLimitException indefinitely and count the number of write attempts within the MockFCS object. Then another test in which MockFCS raises RateLimitException once and then succeeds.

Reply:
Also you can test that exceptions occurred without a try/except by doing:
```
with self.assertRaises(MyExceptionName):
  code_which_raises_the_exception()
```

Reply:
Rewrite the tests. PTAL Thanks